src/ds/ds-motion-common.cpp

```diff
@@ -173,8 +173,10 @@ namespace librealsense
             if (p->get_stream_type() == RS2_STREAM_ACCEL || p->get_stream_type() == RS2_STREAM_GYRO)
             {
                 auto motion = dynamic_cast<motion_stream_profile_interface*>(p.get());
-                assert(motion);
                 auto st = p->get_stream_type();
+
```

Comment:
Better to perform the check immediately after the cast.
1. locality - easier to see the connection
2. efficiency - not performing needless operation in case of failure

Reply:
I will learn it. Fixed.